### PR TITLE
Add check of making `.deb` source package on pull requests

### PR DESCRIPTION
**Pull request by thomasspriggs**
Add check of making `.deb` source package on pull requests

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: cmake/CPackGeneratorConfig.cmake [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# This is run once per generator, which means that generator specific logic can
+# go here.
```
Comment (hannes-steffenhagen-diffblue) [inline cmake/CPackGeneratorConfig.cmake:2]: 👍 sounds like the right approach on principle

**Comment by hannes-steffenhagen-diffblue** [comment]
@thomasspriggs looks like sources for source packages need to be one level deeper, e.g. at the top level of the source package there should only be a single directory containing the source code, I think the name of that is supposed to be `<package name>-<version name>`

File: cmake/CPackGeneratorConfig.cmake [hunk 2]
```diff
@@ -5,5 +5,5 @@ message(STATUS "Configuring for generator ${CPACK_GENERATOR} and TOPLEVEL_TAG ${
 
 if("${CPACK_TOPLEVEL_TAG}" STREQUAL "Linux-Source")
   message(STATUS "Adding python dependency")
-  set(CPACK_DEBIAN_PACKAGE_DEPENDS "${CPACK_DEBIAN_PACKAGE_DEPENDS},python3")
+  set(CPACK_DEBIAN_PACKAGE_DEPENDS "${CPACK_DEBIAN_PACKAGE_DEPENDS},python")
```
Comment (hannes-steffenhagen-diffblue) [inline cmake/CPackGeneratorConfig.cmake]: Can you explain this one? Is this installing both `python3` and `python2`?

Comment (thomasspriggs) [reply cmake/CPackGeneratorConfig.cmake]: The `python` dependency is for python 2.7. The `python3` dependency is for python 3.8. We actually need both to pass lintian, as we have some python scripts which start with `#!/usr/bin/env python` and with `#!/usr/bin/env python3`.

**Pull request closed without merging**